Mention convenient uv commands for installing esphome from git (#61)
esphome/developers.esphome.io: docs/contributing/development-environment.md

        ```bash
        pip install git+https://github.com/username/esphome.git@branch --no-deps --force-reinstall
        ```

>  

Reviewer: ```suggestion

```

        ```

        This will ensure that pip only compiles and reinstalls ESPHome and not its dependencies.

    - If `uv` is installed, which can be done using `pip install uv`, it massively simplifies the above process.
      UV automagically takes care of all the checks using just one command:

        ```bash
        uv pip install git+https://github.com/username/esphome.git@branch --system
        ```

>         The --system flag is only needed for python on Windows installation.

Reviewer: ```suggestion
        The `--system` flag is only necessary when running Python on Windows.
```

Author: Thanks for your patience with my silliness
